bundles/org.openhab.binding.livisismarthome/src/main/java/org/openhab/binding/livisismarthome/internal/handler/LivisiDeviceHandler.java

```diff
@@ -0,0 +1,1003 @@
+            logStateNULL(capability);
+        }
+    }
+
+    private void updateAlarmActuatorChannels(CapabilityDTO capability) {
+        final Boolean alarmState = capability.getCapabilityState().getAlarmActuatorState();
+        if (alarmState != null) {
+            updateState(CHANNEL_ALARM, OnOffType.from(alarmState));
+        } else {
+            logStateNULL(capability);
+        }
+    }
+
+    private void updateMotionDetectionSensorChannels(CapabilityDTO capability) {
+        final Integer motionCount = capability.getCapabilityState().getMotionDetectionSensorState();
+        if (motionCount != null) {
+            logger.debug("Motion state {} -> count {}", motionCount, motionCount);
+            updateState(CHANNEL_MOTION_COUNT, new DecimalType(motionCount));
+        } else {
+            logStateNULL(capability);
+        }
+    }
+
+    private void updateLuminanceSensorChannels(CapabilityDTO capability) {
+        final Double luminance = capability.getCapabilityState().getLuminanceSensorState();
+        if (luminance != null) {
+            updateState(CHANNEL_LUMINANCE, QuantityType.valueOf(luminance, Units.PERCENT));
+        } else {
+            logStateNULL(capability);
+        }
+    }
+
+    private void updatePushButtonSensorChannels(CapabilityDTO capability) {
+        final Integer pushCount = capability.getCapabilityState().getPushButtonSensorCounterState();
+        final Integer buttonIndex = capability.getCapabilityState().getPushButtonSensorButtonIndexState();
+        final String type = capability.getCapabilityState().getPushButtonSensorButtonIndexType();
+        logger.debug("Pushbutton index {}, count {}, type {}", buttonIndex, pushCount, type);
+        if (buttonIndex != null && pushCount != null) {
+            if (buttonIndex >= 0 && buttonIndex <= 7) {
+                final int channelIndex = buttonIndex + 1;
+                if (type != null) {
+                    if (SHORT_PRESS.equals(type)) {
+                        triggerChannel(CHANNEL_BUTTON + channelIndex, CommonTriggerEvents.SHORT_PRESSED);
+                    } else if (LONG_PRESS.equals(type)) {
+                        triggerChannel(CHANNEL_BUTTON + channelIndex, CommonTriggerEvents.LONG_PRESSED);
+                    }
+                }
+                triggerChannel(CHANNEL_BUTTON + channelIndex, CommonTriggerEvents.PRESSED);
+                updateState(String.format(CHANNEL_BUTTON_COUNT, channelIndex), new DecimalType(pushCount));
+            }
+            // Button handled so remove state to avoid re-trigger.
+            capability.getCapabilityState().setPushButtonSensorButtonIndexState(null);
+            capability.getCapabilityState().setPushButtonSensorButtonIndexType(null);
+        } else {
+            logStateNULL(capability);
+        }
+    }
+
+    private void updateEnergyConsumptionSensorChannels(CapabilityDTO capability) {
+        updateStateForEnergyChannelWatt(CHANNEL_ENERGY_CONSUMPTION_MONTH_KWH,
```

Comment:
The method called with provide `Units.WATT`, but actually it should be `Units.KILOWATT_HOUR`. Please check also all channels below ending in **_KWH**.

Reply:
solved